lib/auth.coffee

```diff
@@ -62,7 +80,12 @@ getAuth = (deps, opts) ->
 	# });
 	###
 	exports.whoami = (callback) ->
-		token.getUsername().asCallback(callback)
+		exports.getUserDetails()
+			.then (body) ->
+				return body.username || undefined
+			.catch ->
+				return undefined
```

Comment:
Nit: Bluebird has lots of lovely things. It takes a little while to get used to all of them, but there's a list here: http://bluebirdjs.com/docs/api-reference.html

Notably, you can replace this with just `.catchReturn(undefined)`